test/integration/restart_task_test.rb

```diff
@@ -206,6 +206,37 @@ def test_restart_failure
       in_order: true)
   end
 
+  def test_restart_successful_with_partial_availability
+    result = deploy_fixtures("slow-cloud") do |fixtures|
+      web = fixtures["web.yml.erb"]["Deployment"].first
+      web["spec"]["strategy"]['rollingUpdate']['maxUnavailable'] = '50%'
+      container = web["spec"]["template"]["spec"]["containers"].first
+      container["readinessProbe"] = {
+        "exec" => { "command" => %w(sleep 5) },
+        "timeoutSeconds" => 6
+      }
+    end
+    assert_deploy_success(result)
+
+    restart = build_restart_task
+    assert_restart_success(restart.perform(["web"]))
+
+    pods = kubeclient.get_pods(namespace: @namespace, label_selector: 'name=web,app=slow-cloud')
+    new_pods = pods.select do |pod|
+      pod.spec.containers.select { |c| c["name"] == "app" && c.env&.find { |n| n.name == "RESTARTED_AT" } }
```

Comment:
This will always be truthy, because a `select` with no results is `[]`. So I'd expect that `new_pods` would always include all pods. I'd use `any?` instead.